modules/create_environment_v2/main.tf

```
@@ -0,0 +1,169 @@
+
+      dynamic "ip_allocation_policy" {
+        for_each = (var.pod_ip_allocation_range_name != null || var.service_ip_allocation_range_name != null) ? [1] : []
+        content {
+          cluster_secondary_range_name  = var.pod_ip_allocation_range_name
+          services_secondary_range_name = var.service_ip_allocation_range_name
+        }
+      }
+    }
+
+    dynamic "software_config" {
+      for_each = [
+        {
+          airflow_config_overrides = var.airflow_config_overrides
+          pypi_packages            = var.pypi_packages
+          env_variables            = var.env_variables
+          image_version            = var.image_version
+      }]
+      content {
+        airflow_config_overrides = software_config.value["airflow_config_overrides"]
+        pypi_packages            = software_config.value["pypi_packages"]
+        env_variables            = software_config.value["env_variables"]
+        image_version            = software_config.value["image_version"]
+      }
+    }
+
+    dynamic "private_environment_config" {
+      for_each = var.use_private_environment ? [
+        {
+          enable_private_endpoint                = var.enable_private_endpoint
+          master_ipv4_cidr_block                 = var.master_ipv4_cidr
+          cloud_sql_ipv4_cidr_block              = var.cloud_sql_ipv4_cidr
+          web_server_ipv4_cidr_block             = var.web_server_ipv4_cidr
+          cloud_composer_network_ipv4_cidr_block = var.cloud_composer_network_ipv4_cidr_block
+      }] : []
+      content {
+        enable_private_endpoint                = private_environment_config.value["enable_private_endpoint"]
+        master_ipv4_cidr_block                 = private_environment_config.value["master_ipv4_cidr_block"]
+        cloud_sql_ipv4_cidr_block              = private_environment_config.value["cloud_sql_ipv4_cidr_block"]
+        web_server_ipv4_cidr_block             = private_environment_config.value["web_server_ipv4_cidr_block"]
+        cloud_composer_network_ipv4_cidr_block = private_environment_config.value["cloud_composer_network_ipv4_cidr_block"]
+      }
+    }
+
+    dynamic "maintenance_window" {
+      for_each = (var.maintenance_end_time != null && var.maintenance_recurrence != null) ? [
+        {
+          start_time = var.maintenance_start_time
+          end_time   = var.maintenance_end_time
+          recurrence = var.maintenance_recurrence
+      }] : []
+      content {
+        start_time = maintenance_window.value["start_time"]
+        end_time   = maintenance_window.value["end_time"]
+        recurrence = maintenance_window.value["recurrence"]
+      }
+    }
+
+    workloads_config {
+
+      dynamic "scheduler" {
+        for_each = var.scheduler.enabled ? [
+          {
+            cpu        = var.scheduler.cpu
+            memory_gb  = var.scheduler.memory_gb
+            storage_gb = var.scheduler.storage_gb
+            count      = var.scheduler.count
+        }] : []
```

Do we need `.enabled`? Cant we do something like for some of these blocks like scheduler, web_server, worker etc?

```suggestion
        for_each = var.scheduler != null ? [var.scheduler] : []
```